### PR TITLE
feat(DA): expose configuration options for retry loop backoff

### PR DESCRIPTION
**Pull request by danwt**
# PR Standards

## Opening a pull request should be able to meet the following requirements

---

Closes https://github.com/dymensionxyz/dymint/issues/756 (step 1)

<-- Briefly describe the content of this pull request -->

For Author:

- [ ]  Targeted PR against correct branch
- [ ]  included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Linked to Github issue with discussion and accepted design
- [ ]  Targets only one github issue
- [ ]  Wrote unit and integration tests
- [ ]  All CI checks have passed
- [ ]  Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code)

---

For Reviewer:

- [ ]  confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Reviewers assigned
- [ ]  confirmed all author checklist items have been addressed

---

After reviewer approval:

- [ ]  In case targets main branch, PR should be squashed and merged.
- [ ]  In case PR targets a release branch, PR should be rebased.

### CODE REVIEW

**Review by omritoptix — CHANGES_REQUESTED** [review]
I'm not fining a reason to not put all the time the `defaultRpcCheckAttempts` on celestia to 0 (i.e try forever) as both states of retrieveal/submission assuming it's not a config error.

File: config/toml.go [hunk 1]
```diff
@@ -79,6 +79,7 @@ batch_submit_max_time = "{{ .BlockManagerConfig.BatchSubmitMaxTime }}"
 ### da config ###
 da_layer = "{{ .DALayer }}" # mock, celestia, avail
 namespace_id = "{{ .BlockManagerConfig.NamespaceID }}"
+# this should be json matching the celestia.Config type
```
Comment (mtsitrin) [inline config/toml.go:82]: pls update the `#celestia config example:` below as well

**Review by mtsitrin — CHANGES_REQUESTED** [review]

File: da/celestia/celestia.go [hunk 2]
```diff
@@ -111,6 +94,33 @@ func (c *DataAvailabilityLayerClient) Init(config []byte, pubsubServer *pubsub.S
...
+	err = c.InitNamespaceID()
+	if err != nil {
+		return c, fmt.Errorf("init namespace id: %w", err)
+	}
+
+	if c.GasPrices == 0 {
+		return c, errors.New("gas prices must be set")
+	}
+
+	if c.RetryDelay == 0 {
```
Comment (mtsitrin) [inline da/celestia/celestia.go:115]: missing default for `RetryAttempts`

**Review by mtsitrin — CHANGES_REQUESTED** [review]

File: da/celestia/celestia.go (hunk 2 above)
Comment (danwt) [reply da/celestia/celestia.go:115]: its not missing, 0 is a valid value so you cant disambiguate

**Comment by danwt** [comment]
@omritoptix 
>I'm not fining a reason to not put all the time the defaultRpcCheckAttempts on celestia to 0 (i.e try forever) as both states of retrieveal/submission assuming it's not a config error.

That's fine, that's how the PR is actually

File: da/celestia/celestia.go (hunk 2 above)
Comment (mtsitrin) [reply da/celestia/celestia.go:117]: why not setting `defaultRetryAttempts` to be `0`?
I think it's more explicit

**Review by mtsitrin — APPROVED** [review]

**Pull request merged**